sdk/storage/storage-file-datalake/src/AccountSASPermissions.ts

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+/**
+ * ONLY AVAILABLE IN NODE.JS RUNTIME.
+ *
+ * This is a helper class to construct a string representing the permissions granted by an AccountSAS. Setting a value
+ * to true means that any SAS which uses these permissions will grant permissions for that operation. Once all the
+ * values are set, this should be serialized with toString and set as the permissions field on an
+ * {@link AccountSASSignatureValues} object. It is possible to construct the permissions string without this class, but
+ * the order of the permissions is particular and this class guarantees correctness.
+ *
+ * @export
+ * @class AccountSASPermissions
+ */
+export class AccountSASPermissions {
+  /**
+   * Parse initializes the AccountSASPermissions fields from a string.
+   *
+   * @static
+   * @param {string} permissions
+   * @returns {AccountSASPermissions}
+   * @memberof AccountSASPermissions
+   */
+  public static parse(permissions: string): AccountSASPermissions {
+    const accountSASPermissions = new AccountSASPermissions();
+
+    for (const c of permissions) {
+      switch (c) {
+        case "r":
+          accountSASPermissions.read = true;
+          break;
+        case "w":
+          accountSASPermissions.write = true;
+          break;
+        case "d":
+          accountSASPermissions.delete = true;
+          break;
+        case "l":
+          accountSASPermissions.list = true;
+          break;
+        case "a":
+          accountSASPermissions.add = true;
+          break;
+        case "c":
+          accountSASPermissions.create = true;
+          break;
+        case "u":
+          accountSASPermissions.update = true;
+          break;
+        case "p":
+          accountSASPermissions.process = true;
+          break;
+        default:
+          throw new RangeError(`Invalid permission character: ${c}`);
+      }
+    }
+
+    return accountSASPermissions;
+  }
+
+  /**
+   * Permission to read resources and list queues and tables granted.
+   *
+   * @type {boolean}
+   * @memberof AccountSASPermissions
+   */
+  public read: boolean = false;
+
+  /**
+   * Permission to write resources granted.
+   *
+   * @type {boolean}
+   * @memberof AccountSASPermissions
+   */
+  public write: boolean = false;
+
+  /**
+   * Permission to create blobs and files granted.
+   *
+   * @type {boolean}
+   * @memberof AccountSASPermissions
+   */
+  public delete: boolean = false;
+
+  /**
+   * Permission to list blob containers, blobs, shares, directories, and files granted.
+   *
+   * @type {boolean}
+   * @memberof AccountSASPermissions
+   */
+  public list: boolean = false;
+
+  /**
+   * Permission to add messages, table entities, and append to blobs granted.
+   *
+   * @type {boolean}
+   * @memberof AccountSASPermissions
+   */
+  public add: boolean = false;
+
+  /**
+   * Permission to create blobs and files granted.
+   *
```

Comment:
General which applies to other fields, what's the document to support adls gen 2 account SAS. Existing things are related to blob. And this SDK is possibly for data lake user only?

Reply:
There is no document. Basically, dfs endpoint applies all blob sas.